### PR TITLE
Add more tests for editing version license and compatibility

### PR DESCRIPTION
**Pull request by AlexandraMoga**
This PR includes:
- tests for editing a version custom license (valid and invalid cases)
- tests for editing a version compatibility (valid and invalid cases)

### CODE REVIEW

File: tests/api/test_addon_uploads.py [hunk 1]
```diff
@@ -1922,6 +1953,285 @@ def test_edit_version_both_license_and_custom_license(base_url, session_auth):
...
+        (
+            ['android', 'firefox'],
+            {
+                'android': {'min': '48.0', 'max': '*'},
+                'firefox': {'min': '42.0', 'max': '*'},
+            },
+        ),
+        (['firefox'], {'firefox': {'min': '42.0', 'max': '*'}}),
+        ({'firefox': {'min': '65.0'}}, {'firefox': {'min': '65.0', 'max': '*'}}),
+        ({'android': {'max': '95.0'}}, {'android': {'min': '48.0', 'max': '95.0'}}),
```
Comment (jrbenny35) [inline tests/api/test_addon_uploads.py:2130]: What is going on here exactly?

File: tests/api/test_addon_uploads.py [hunk 2]
```diff
@@ -1922,6 +1953,285 @@ def test_edit_version_both_license_and_custom_license(base_url, session_auth):
...
+        url=f'{base_url}{_addon_create}{addon}/versions/{version}/',
+        headers={
+            'Authorization': f'Session {session_auth}',
+            'Content-Type': 'application/json',
+        },
+        data=json.dumps(payload),
+    )
+    print(
+        f'For compatibility "{value}": Response status is '
+        f'{edit_version.status_code}; {edit_version.text}\n'
```
Comment (jrbenny35) [inline tests/api/test_addon_uploads.py:2215]: Is this supposed to be here?

File: tests/api/test_addon_uploads.py [hunk 3]
```diff
@@ -1922,6 +1953,285 @@ def test_edit_version_both_license_and_custom_license(base_url, session_auth):
...
+        url=f'{base_url}{_addon_create}{addon}/versions/{version}/',
+        headers={
+            'Authorization': f'Session {session_auth}',
+            'Content-Type': 'application/json',
+        },
+        data=json.dumps(payload),
+    )
+    print(
+        f'For custom_license "{value}": Response status is '
+        f'{edit_version.status_code}; {edit_version.text}\n'
```
Comment (jrbenny35) [inline tests/api/test_addon_uploads.py:1995]: Is this supposed to be here?

**Review by jrbenny35 — CHANGES_REQUESTED** [review]
Looking good, just some questions. I see some print statements in the code, are those needed?

File: tests/api/test_addon_uploads.py (hunk 3 above)
Comment (AlexandraMoga) [reply tests/api/test_addon_uploads.py:1995]: I left the print there because it helps me debug failures more easily by checking the response code and the response message before the actual validations (assertions). 
I could add all of this in the first assert message (the one immediately following) as an alternative solution.

File: tests/api/test_addon_uploads.py [hunk 4]
```diff
@@ -1922,6 +1953,285 @@ def test_edit_version_both_license_and_custom_license(base_url, session_auth):
...
+        edit_version.status_code == 400
+    ), f'Actual status code was {edit_version.status_code}'
+    if value is None:
+        assert (
+            'This field may not be null.' in edit_version.text
+        ), f'Actual response message was {edit_version.text}'
+    else:
+        assert (
+            'Invalid data. Expected a dictionary' in edit_version.text
+        ), f'Actual response message was {edit_version.text}'
```
[the remark this answers is not in the document]
Comment (AlexandraMoga) [reply tests/api/test_addon_uploads.py:2007]: >The else statement will always run..

Except when the value is `None`,  in which case the else statement will not run (confirmed through executing the code line by line);

>you are expecting a 'Invalid data. Expected a dictionary' message?

Yes, the `custom_license` field is expecting data in dictionary format, so any other data (except `None`) should return this response message.

File: tests/api/test_addon_uploads.py (hunk 1 above)
Comment (AlexandraMoga) [reply tests/api/test_addon_uploads.py:2130]: Yes, the parameters are a bit hard to follow indeed. I've tried to make the `ids` suggestive but I guess that without knowing how the API works it's not all that clear.

So, each parameter contains a valid `compatibility` `request_value` - for example, in the first parameter we send `['android', 'firefox']` as compatibility, which should be interpreted by the API as `compatibility: {'android': {'min': '48.0', 'max': '*'}, 'firefox': {'min': '42.0', 'max': '*'}}`, which is what I've set as the `response_value` parameter to compare against. 

In  the  third parameter, I'm setting the `request_value` as `{'firefox': {'min': '65.0'}}`, which means I'm making my addon compatible with Firefox 65.0 as min value and the API response I'm expecting is `compatibility:  {'firefox': {'min': '65.0', 'max': '*'}}`.

Hope this clarifies things a bit.

File: tests/api/test_addon_uploads.py (hunk 2 above)
Comment (AlexandraMoga) [reply tests/api/test_addon_uploads.py:2215]: I left the print there because it helps me debug failures more easily by checking the response code and the response message before the actual validations (assertions).
I could add all of this in the first assert message (the one immediately following) as an alternative solution.

Comment (jrbenny35) [reply tests/api/test_addon_uploads.py:2215]: Well if you think it's useful I don't mind, I do wonder if the assert message idea would be a bit cleaner. Up to you though.

**Review by jrbenny35 — APPROVED** [review]
